primitives/src/signatures/bls.rs

```diff
@@ -89,15 +48,13 @@ where
         _pp: &Self::PublicParameter,
         prng: &mut R,
     ) -> Result<(Self::SigningKey, Self::VerificationKey), PrimitivesError> {
-        // TODO: absorb ciphersuite_id in prng
-        let sk = BLSSignKey(<P::G1Parameters as ModelParameters>::ScalarField::rand(
-            prng,
-        ));
-        let vk = BLSVerKey(
-            GroupAffine::<P::G2Parameters>::prime_subgroup_generator()
-                .mul(sk.0)
-                .into_affine(),
-        );
+        let mut ikm = [0u8; 32];
+        prng.fill_bytes(&mut ikm);
+        let sk = match SecretKey::key_gen(&ikm, &[]) {
+            Ok(sk) => sk,
+            Err(e) => return Err(PrimitivesError::InternalError(format!("{:?}", e))),
```

Comment:
+80

can we add a wrapper in `error.rs` and define a new error type for this?

Reply:
Here the error shall never occur. The only case where it will happen, is when the length of ikm is less than 32.